server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/HTTPConfigurationServerTest.java

```diff
@@ -0,0 +1,130 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mock.smtp.server;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.RestAssured.when;
+import static io.restassured.RestAssured.with;
+import static io.restassured.config.EncoderConfig.encoderConfig;
+import static io.restassured.config.RestAssuredConfig.newConfig;
+import static net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson;
+import static org.apache.james.mock.smtp.server.Fixture.JSON_BEHAVIORS;
+import static org.hamcrest.Matchers.hasSize;
+
+import java.nio.charset.StandardCharsets;
+
+import org.eclipse.jetty.http.HttpStatus;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.restassured.RestAssured;
+import io.restassured.builder.RequestSpecBuilder;
+import io.restassured.http.ContentType;
+import net.javacrumbs.jsonunit.core.Option;
+import net.javacrumbs.jsonunit.core.internal.Options;
+
+class HTTPConfigurationServerTest {
+    private HTTPConfigurationServer server;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        server = HTTPConfigurationServer.onRandomPort(new SMTPBehaviorRepository());
+        server.start();
+
+        RestAssured.requestSpecification = new RequestSpecBuilder()
+            .setContentType(ContentType.JSON)
+            .setAccept(ContentType.JSON)
+            .setConfig(newConfig().encoderConfig(encoderConfig().defaultContentCharset(StandardCharsets.UTF_8)))
+            .setPort(server.getPort().getValue())
+            .setBasePath("/")
+            .setBasePath("/smtpBehaviors")
+            .build();
+    }
+
+    @AfterEach
+    void tearDown() throws Exception {
+        server.stop();
+    }
+
+    @Test
+    void getShouldReturnEmptyByDefault() {
+        when()
+            .get()
+        .then()
+            .body(".", hasSize(0));
+    }
+
+    @Test
+    void getShouldReturnPreviouslyStoredData() {
+        with().body(JSON_BEHAVIORS).put();
+
+        String response = when()
+            .get()
+            .then()
+            .extract().asString();
+
+        assertThatJson(response)
+            .withOptions(new Options(Option.TREATING_NULL_AS_ABSENT, Option.IGNORING_ARRAY_ORDER))
+            .isEqualTo(JSON_BEHAVIORS);
+    }
+
+    @Test
+    void getShouldReturnEmptyAfterDelete() {
+        with().body(JSON_BEHAVIORS).put();
+
+        with().delete();
+
+        when()
+            .get()
+        .then()
+            .body(".", hasSize(0));
+    }
+
+    @Test
+    void putShouldReturnNoContent() {
+        given()
+            .body(JSON_BEHAVIORS)
+        .when()
+            .put()
```

Comment:
In the gitlab tickets you used the HTTP verb `POST` which seemed to be the right choice given that the resource is not identified in the URI. Why did you choose to use the `PUT` verb ?

Reply:
Ok it's for updating the list of behavior so i guess `PUT` is the right choice. Could you just update the gitlab ticket to be coherent with your new decision ?

Reply:
> Ok it's for updating the list of behavior and it's idempotent so PUT is the right choice. 

That's it!

> Could you just update the gitlab ticket to be coherent with your new decision ?

Sure <3